datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -155,7 +157,8 @@ fn roundtrip_statement() -> Result<()> {
 
         let context = MockContextProvider::default()
             .with_udaf(sum_udaf())
-            .with_udaf(count_udaf());
+            .with_udaf(count_udaf())
+            .with_expr_planner(Arc::new(CoreFunctionPlanner::default()));
```

Comment:
I prefer not to re-introduce `with_expr_planner` back for context provider.

It is better to use session state in context, and add planner into session state.

Similar to SessionContextProvider
```rust
struct SessionContextProvider<'a> {
    state: &'a SessionState,
    ...
}
```

Reply:
How about we move this change altogether to a new issue ? I see that we have to update many place. Also we can remove `with_udaf` and make it consistent across places.

Reply:
Do you mean open a new PR? Sure, if so.

Reply:
Created Issue https://github.com/apache/datafusion/issues/11531 to track this.